physionet-django/project/models.py

```diff
@@ -1533,28 +1533,179 @@ def get_storage_info(self, force_calculate=True):
+                        authors=temp_authors,
+                        year=self.publish_datetime.year,
+                        title=self.title,
+                        version=self.version)
+
+            elif (style == 'Chicago'):
+
+                if (len(authors) == 1):
+                    temp_authors = authors[0].get_full_name(reverse=True) 
+                else:
+                    temp_authors = ', '.join(a.get_full_name(reverse=True) for a in authors[:(len(authors)-1)])
+                    temp_authors += ', and ' + authors[len(authors)-1].get_full_name()
+
+                if self.doi:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}). '
+                        '<a href="https://doi.org/{doi}">https://doi.org/{doi}</a>.',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year,
+                        doi=self.doi)
+                else:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}).',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year)
+
+            elif (style == 'Harvard'):
+
+                if (len(authors) == 1):
+                    temp_authors = authors[0].initialed_name() 
+                else:
+                    temp_authors = ', '.join(a.initialed_name() for a in authors[:(len(authors)-1)])
+                    temp_authors += ', and ' + authors[len(authors)-1].initialed_name()
+
+                if self.doi:
+                    citation_format = format_html(
+                        "{authors} ({year}) '{title} (version {version}).', "
+                        "<i>PhysioNet</i>, Available from: "
+                        "<a href='https://doi.org/{doi}'>https://doi.org/{doi}</a>.",
+                        authors=temp_authors,
+                        year=self.publish_datetime.year,
+                        title=self.title,
+                        version=self.version,
+                        doi=self.doi)
+                else:
+                    citation_format = format_html(
+                        "{authors} ({year}) '{title} (version {version}).', "
+                        "<i>PhysioNet</i>.",
+                        authors=temp_authors,
+                        year=self.publish_datetime.year,
+                        title=self.title,
+                        version=self.version)
+
+            elif (style == 'Vancouver'):
+
+                temp_authors = ', '.join(a.initialed_name().replace(',','').replace('.','') for a in authors)
```

Comment:
I think PEP8 states <80 characters in width.

Reply:
Good catch